endstate_correction/protocol.py

```diff
@@ -261,26 +268,26 @@ def perform_endstate_correction(protocol: Protocol) -> Results:
             print("Performing bidirectional protocol...")
             lambs = np.linspace(0, 1, protocol.neq_switching_length)
             Ws, endstates_mm_to_qml, trajs_mm_to_qml = perform_switching(
-                    sim,
-                    lambs,
-                    samples=protocol.trajectories[0],
-                    nr_of_switches=protocol.nr_of_switches,
-                    save_endstates=protocol.save_endstates,
-                    save_trajs=protocol.save_trajs
-                )
-            Ws_from_mm_to_qml = np.array(Ws/ kBT)
+                sim,
+                lambs,
+                samples=protocol.trajectories[0],
+                nr_of_switches=protocol.nr_of_switches,
+                save_endstates=protocol.save_endstates,
+                save_trajs=protocol.save_trajs,
+            )
+            Ws_from_mm_to_qml = np.array(Ws / kBT)
 
             # perform switching from qml to mm
             lambs = np.linspace(1, 0, protocol.neq_switching_length)
             Ws, endstates_qml_to_mm, trajs_qml_to_mm = perform_switching(
-                    sim,
-                    lambs,
-                    samples=protocol.trajectories[-1],
-                    nr_of_switches=protocol.nr_of_switches,
-                    save_endstates=protocol.save_endstates,
-                    save_trajs=protocol.save_trajs
-                )
-            Ws_from_qml_to_mm = np.array(Ws/ kBT)   
+                sim,
+                lambs,
+                samples=protocol.trajectories[1],
```

Comment:
Why the change from `samples=protocol.trajectories[-1]` to `samples=protocol.trajectories[1]`

Reply:
this is not very verbose and is a potential point of failure or ambiguity.
There are two "sample pools" generated with two different Hamiltonians. Both of them are in the list. So `-1` is equivalent to `1`. but I will try to come up with a non-ambiguous way to solve this.

Reply:
I see. Maybe make this into a dictionary instead of a list `protocol.trajectories=dict(forward=AAA, backward=BBB)`?

Reply:
I will add separate fields in the dataclass for reference/target samples. That might be the most verbose approach.
Thanks for raising this.